tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -58,66 +77,78 @@ func (r *Report) writeToArtifactsDir() error {
+		if err != nil {
+			return err
+		}
+		return r.GetFlakyTestReport(repo, latestBuild)
 	}
-	return getLatestValidJSON(job, relPath)
+	build := job.NewBuild(buildID)
+	for _, filepath := range getReportPaths(build, repo) {
+		report, err := readJSONReport(build, filepath)
+		if err != nil {
+			return err
+		}
+		r.AllReports = append(r.AllReports, *report)
+	}
+	return nil
 }
 
-// inexpensively sort and find the most recent JSON file. Assumes build IDs are in sequential order.
-func getLatestValidJSON(job *prow.Job, relPath string) ([]byte, error) {
-	maxElapsedTime, _ := time.ParseDuration(fmt.Sprintf("%dh", maxAge*24))
-	buildIDs := job.GetBuildIDs()
-	sort.Sort(sort.Reverse(sort.IntSlice(buildIDs)))
-	for _, buildID := range buildIDs {
-		build := job.NewBuild(buildID)
-		// check for JSON
-		if contents, err := build.ReadFile(relPath); err != nil {
-			continue
-		} else {
-			// JSON exists, check timestamp
-			startTimeInt, err := build.GetStartTime()
-			if err != nil {
-				continue
-			}
-			startTime := time.Unix(startTimeInt, 0)
-			if time.Since(startTime) < maxElapsedTime {
-				return contents, nil
-			} else {
-				return nil, fmt.Errorf("latest JSON log is outdated: %.2f days old", time.Since(startTime).Hours()/24)
-			}
+// getReportPaths searches build artifacts for reports from the given repo, returning
+// the path to any matching files. Use repo = "" to get all reports from all repos.
+func getReportPaths(build *prow.Build, repo string) []string {
+	var matches []string
+	suffix := path.Join(repo, filename)
+	for _, artifact := range build.GetArtifacts() {
+		if strings.HasSuffix(artifact, suffix) {
+			matches = append(matches, artifact)
 		}
 	}
-	return nil, fmt.Errorf("no JSON logs found in recent builds")
+	return matches
 }
 
-// GetReportForRepo gets a Report struct from a specific build for a given repo
-// use buildID = -1 for most recent successful report
-func GetReportForRepo(repo string, buildID int) (*Report, error) {
-	report := &Report{
-		Repo: repo,
-	}
-	contents, err := getJSONForBuild(repo, buildID)
+// readJSONReport builds a repo-specific report object from a given json file path.
+func readJSONReport(build *prow.Build, filename string) (*RepoReport, error) {
+	report := &RepoReport{}
+	contents, err := build.ReadFile(filename)
 	if err != nil {
 		return nil, err
 	}
-	if err = json.Unmarshal(contents, report); err != nil {
+	if err := json.Unmarshal(contents, report); err != nil {
 		return nil, err
 	}
 	return report, nil
 }
 
-// CreateReportForRepo generates a Report struct and optionally writes it to disk
-func CreateReportForRepo(repo string, flaky []string, writeFile bool) (*Report, error) {
-	report := &Report{
-		Repo:  repo,
-		Flaky: flaky,
-	}
-	if writeFile {
-		return report, report.writeToArtifactsDir()
+// getLatestValidBuild inexpensively sorts and finds the most recent JSON report.
+// Assumes build IDs are in sequential order.
```

Comment:
why is this an assumption? You are sorting it anyways?

Reply:
I worded that comment poorly. Assume sequential build IDs are sequential in time, i.e. if latest is ID 3, the previous build is 2, the one before that is 1, etc. Will fix.